### PR TITLE
cephadm: rewrite call() with asyncio

### PR DESCRIPTION
**Pull request by tchaikov**
for better readability

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test api

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -1200,117 +1235,53 @@ def call(ctx: CephadmContext,
...
+            protocol_factory,
+            *command,
+            close_fds=True,
+            **kwargs)
+        proc_transport = cast(asyncio.SubprocessTransport, transport)
+        proc_protocol = cast(StreamReaderProto, protocol)
+        returncode = 0
+        try:
+            if timeout:
+                await asyncio.wait_for(proc_exited, timeout)
```
Comment (sebastian-philipp) [inline src/cephadm/cephadm:1261]: according to the docs, this will also raise a timeout, if the process is blocked. This was previously not the case.

**Review by sebastian-philipp — APPROVED** [review]
looks good! 

fyi, in the past we had issues with:

* race between the process is already finished while we still read stdout
* data too big for buffers
* data not being printed while the process is still running.

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2021-01-23_14:54:00-rados-wip-kefu-testing-2021-01-23-2058-distro-basic-smithi/

**Pull request merged**